src/test/java/org/broadinstitute/hellbender/tools/copynumber/PreprocessIntervalsIntegrationTest.java

```diff
@@ -0,0 +1,146 @@
+        final ArrayList<Interval> inputIntervalsOverlappingIntervalTest = new ArrayList<Interval>();
+        inputIntervalsOverlappingIntervalTest.add(new Interval("20", 3_000, 20_000));
+        inputIntervalsOverlappingIntervalTest.add(new Interval("20", 200, 2_100));
+        final ArrayList<Interval> expectedBinsOverlappingIntervalTest = new ArrayList<Interval>();
+        expectedBinsOverlappingIntervalTest.add(new Interval("20", 1, 10_000));
+        expectedBinsOverlappingIntervalTest.add(new Interval("20", 10_001, 20_000));
+        expectedBinsOverlappingIntervalTest.add(new Interval("20", 20_001, 20_500));
+
+        // Test for intervals reaching the ends of the contigs
+        final int binLengthEdgeIntervalTest = 10_000;
+        final int paddingEdgeIntervalTest = 500;
+        final String[] outputFileNameEdgeIntervalTest =  {"GATK-preprocess-intervals-test", ".tmp"};
+        final ArrayList<Interval> inputIntervalsEdgeIntervalTest = new ArrayList<Interval>();
+        inputIntervalsEdgeIntervalTest.add(new Interval("20", 3000, 20000));
+        inputIntervalsEdgeIntervalTest.add(new Interval("20", 63025220, 63025520));
+        final ArrayList<Interval> expectedBinsEdgeIntervalTest = new ArrayList<Interval>();
+        expectedBinsEdgeIntervalTest.add(new Interval("20", 2500, 12499));
+        expectedBinsEdgeIntervalTest.add(new Interval("20", 12500, 20500));
+        expectedBinsEdgeIntervalTest.add(new Interval("20", 63024720, 63025520));
+
+        // Test for whole chromosome
+        final int binLengthWholeChromosomeTest = 10_000_000;
+        final int paddingWholeChromosomeTest = 500;
+        final String[] outputFileNameWholeChromosomeTest =  {"GATK-preprocess-intervals-test", ".tmp"};
+        final ArrayList<Interval> inputIntervalsWholeChromosomeTest = new ArrayList<Interval>();
+        inputIntervalsWholeChromosomeTest.add(new Interval("20", 1, 63_025_520));
+        final ArrayList<Interval> expectedBinsWholeChromosomeTest = new ArrayList<Interval>();
+        expectedBinsWholeChromosomeTest.add(new Interval("20", 1, 10_000_000));
+        expectedBinsWholeChromosomeTest.add(new Interval("20", 10_000_001, 20_000_000));
+        expectedBinsWholeChromosomeTest.add(new Interval("20", 20_000_001, 30_000_000));
+        expectedBinsWholeChromosomeTest.add(new Interval("20", 30_000_001, 40_000_000));
+        expectedBinsWholeChromosomeTest.add(new Interval("20", 40_000_001, 50_000_000));
+        expectedBinsWholeChromosomeTest.add(new Interval("20", 50_000_001, 60_000_000));
+        expectedBinsWholeChromosomeTest.add(new Interval("20", 60_000_001, 63_025_520));
+
+        // Test for whole genome -- when we don't give any intervals, then the tool assumes that the user wants to sequence the whole genome
+        final int binLengthWholeGenomeTest = 10_000_000;
+        final int paddingWholeGenomeTest = 500;
+        final String[] outputFileNameWholeGenomeTest =  {"GATK-preprocess-intervals-test", ".tmp"};
```

Comment:
this name is the same across all test cases so you can just move it into the test itself

Reply:
:+1:

Reply:
OK